.ci/no-exception-test.sh

```diff
@@ -54,22 +54,24 @@ guava-with-sun-checks)
   ;;
 
 openjdk14-with-checks-nonjavadoc-error)
-  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
-                     --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
-  echo CS_version: $CS_POM_VERSION
+  LOCAL_GIT_REPO=$(pwd)
+  BRANCH=$(git rev-parse --abbrev-ref HEAD)
   mkdir -p .ci-temp/
   cd .ci-temp/
   git clone https://github.com/checkstyle/contribution
   cd ..
   sed -i.'' 's/value=\"error\"/value=\"ignore\"/' \
         .ci-temp/contribution/checkstyle-tester/checks-nonjavadoc-error.xml
   cd .ci-temp/contribution/checkstyle-tester
+  cp ../../../.ci/openjdk-projects-to-test-on.config openjdk-projects-to-test-on.config
   sed -i '/  <!-- Filters -->/r ../../../.ci/openjdk14-excluded.files' checks-nonjavadoc-error.xml
   export MAVEN_OPTS="-Xmx2048m"
-  groovy ./launch.groovy --listOfProjects ../../../.ci/openjdk-projects-to-test-on.config \
-      --config checks-nonjavadoc-error.xml \
-      --checkstyleVersion $CS_POM_VERSION \
-      --extraMvnOptions "\"-Dmaven.jxr.skip=true\""
+  groovy ./diff.groovy --listOfProjects openjdk-projects-to-test-on.config \
```

Comment:
@nmancus1 @romani Did we test that diff.groovy will behave the same as launch for no exception testing? We have an issue still open in contribution that we did NOT add this functionality to `diff.groovy` to support no-exception or no-error testing. Diff's main purpose is to continue no matter what.

See https://github.com/checkstyle/contribution/issues/529
> The only thing launch truthfully does now is no-error and no-exception testing. https://github.com/checkstyle/checkstyle/blob/master/.ci/no-exception-test.sh#L26-L27 . To support this, diff needs support to not-ignore the exclude paths as it is common to ignore java files that are only made for testing. Diff doesn't honor excludes by default.

Reply:
From the first line of the PR, it looks like we should have only removed JXR usage and not switched to diff.groovy.

Reply:
>  Did we test that diff.groovy will behave the same as launch for no exception testing?

From what I can tell after working with diff is that this property is derived from the config itself, using the `haltOnException` property.

Here are two examples showing that exception still causes failure of diff IF `haltOnException` is not set to `false`:
successful completion - https://gist.github.com/nmancus1/0a290ac44a4b1badda82017795579e10

failure - https://gist.github.com/nmancus1/2e9649223b08f42a7c93d2adb268df0b

Reply:
Put it simply, launch cannot easily be replaced with diff until the issue mentioned in the my first post. Seeing this happen early before the issue was done raised concerns hence why I was asking about testing.

> haltOnException 

This property is one part of it. The other is `failsOnError`, https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/pom.xml#L18 , which diff turns off. I don't remember anymore who handles what exactly, but both are needed for correct functionality. diff doesn't set it to the right value as far as I know.

I can't say there is no issues with this PR as it is now, unless testing shows otherwise. For example, was the exit code of diff examined when it finished failed regressions. It needs to send the correct exit code otherwise the CI won't see it as erroring and will incorrectly assume it passed.

Also diff skips over the ignore list, so this brings up 2 conflicting issues. If the run is fine, why do we have the ignore list, if the run isn't fine then how come it passes and was merged.

> failure - https://gist.github.com/nmancus1/2e9649223b08f42a7c93d2adb268df0b

Ok, so it failed on an exception. This is a no exception test so it shouldn't be a big issue then. However as it stands now, it won't produce a failure if there is a violation as diff is made to report violations not fail on them.

We can move on if things look good, but there shouldn't be anymore conversions until the merge issue is done.